### ISSUE TITLE
deheader v0.6 does not handle C++ source codes which read from stdin

### ISSUE THREAD

**Issue by evandrix** [yourabi/deheader#1]
fix: maybe freopen stdin from file?

**Comment by yourabi** [comment]
@evandrix hi - can you be a bit more specific - when you say "does not handle" not sure what you mean... also can you attach some sample code with the error.

**Comment by yourabi** [comment]
Passing on this until failure case is reproducible.

**Comment by evandrix** [comment]
When I say "does not handle", I mean it just freezes waiting for standard input to be supplied by the user via stdin, instead of skipping ahead past these lines to detect which C++ header includes are required for the given file.

An example would be some sample C++ code for competitive programming, where the program is usually given a set of test cases, to print an output answer for each of these input test cases.

In this scenario, I'm unable to run your tool through my C++ code, without first supplying an input to the program (which I can't, unless I compile the code first)

Basically I'm referring to C++ programs containing printf/fprintf() calls.

**Comment by evandrix** [comment]
Please reopen this issue if sufficient details are provided, if not i'll open one myself.

**Comment by yourabi** [comment]
Hi --

First of all I'm not the author of this -- as I state on the project page (and in the Readme) this is a github clone since this isn't in source control anywhere.

I asked you to attach code that was failing -- I wasn't asking for some theoretical example. It's pretty impossible to debug from some vague description instead of a reproducible error.

I tested  against this follow test file (since you mentioned printf).

< / > removed for github
#include stdio.h
#include iostream
#include string

using namespace std;

int main()
{
  cout << "test1" << endl;
  printf ("hello world...\n");
  return 0;
}

And deheader worked perfectly....

git clone ... <deheader.git> ...
deheader/deheader 
deheader: remove <string> from ./test.cpp
deheader: saw 1 files, 3 includes, 1 removable

So as it stands (since you haven't provided any sample code that fails) I think you're doing something wrong here...
